internal/metrics/metrics.go

```diff
@@ -0,0 +1,83 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type tracker interface {
+	TrackRequest(service, method string, status int, duration time.Duration)
+	AddInflightRequest(service string)
+	SubtractInflightRequest(service string)
+}
+
+var Tracker tracker = &nullTracker{}
+
+func Enable() http.Handler {
+	Tracker = NewPrometheusTracker()
+	return promhttp.Handler()
+}
+
+type nullTracker struct{}
+
+func (nullTracker) TrackRequest(service, method string, status int, dur time.Duration) {}
+func (nullTracker) AddInflightRequest(service string)                                  {}
+func (nullTracker) SubtractInflightRequest(service string)                             {}
+
+type prometheusTracker struct {
+	httpRequests     *prometheus.CounterVec
+	httpDuration     *prometheus.HistogramVec
+	inflightRequests *prometheus.GaugeVec
+}
+
+func NewPrometheusTracker() *prometheusTracker {
+	tracker := &prometheusTracker{
+		httpRequests: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "http_requests_total",
+				Help: "HTTP requests processed, labeled by service, status code and method.",
+			},
+			[]string{"service", "method", "status"},
```

Comment:
Just checking: is Kamal doing any input validation on the `method` label (constraining it to reasonable values) before it gets used for the metric here? Otherwise a malicious user could flood the proxy with requests that cycle through a million made-up method names, generating a metric for each of them that sticks around forever (causing ever-increasing memory usage in the Kamal proxy and potentially overloading the Prometheus server with the resulting cardinality explosion).